loopy/schedule/checker/schedule.py

```diff
@@ -214,36 +341,411 @@ def generate_pairwise_schedules(
+                assert isinstance(
+                    lin_item, (CallKernel, ReturnFromKernel))
+                pass
+
+        blex_order_map_params = sorted(blex_order_map_params)
+
+        # At this point, some blex tuples may have more dimensions than others;
+        # the missing dims are the fastest-updating dims, and their values should
+        # be zero. Add them.
+        for stmt, tup in stmt_inst_to_blex.items():
+            stmt_inst_to_blex[stmt] = _pad_tuple_with_zeros(tup, n_seq_blex_dims)
+
+        # }}}
+
+        # {{{ Second, create the blex order map
+
+        # {{{ Create the initial (pre-subtraction) blex order map
+
+        # Create names for the blex dimensions for sequential loops
+        seq_blex_dim_names = [
+            LEX_VAR_PREFIX+str(i) for i in range(n_seq_blex_dims)]
+        seq_blex_dim_names_prime = append_mark_to_strings(
+            seq_blex_dim_names, mark=BEFORE_MARK)
+
+        # Begin with the blex order map created as a standard lexicographical order
+        blex_order_map = create_lex_order_map(
+            dim_names=seq_blex_dim_names,
+            in_dim_mark=BEFORE_MARK,
+            )
+
+        # Add LID/GID dims to blex order map
+        blex_order_map = add_and_name_isl_dims(
+            blex_order_map, dt.out, all_par_lex_dim_names)
+        blex_order_map = add_and_name_isl_dims(
+            blex_order_map, dt.in_,
+            append_mark_to_strings(all_par_lex_dim_names, mark=BEFORE_MARK))
+        if sync_kind == "local":
+            # For intra-group case, constrain GID 'before' to equal GID 'after'
+            for var_name in gid_lex_dim_names:
+                blex_order_map = add_eq_isl_constraint_from_names(
+                        blex_order_map, var_name, var_name+BEFORE_MARK)
+        # (if sync_kind == "global", don't need constraints on LID/GID vars)
+
+        # }}}
+
+        # {{{ Subtract unwanted pairs from happens-before blex map
+
+        # Create map from iname to corresponding blex dim name
+        iname_to_blex_var = {}
+        for iname, dim in iname_to_blex_dim.items():
+            iname_to_blex_var[iname] = seq_blex_dim_names[dim]
+            iname_to_blex_var[iname+BEFORE_MARK] = seq_blex_dim_names_prime[dim]
+
+        # Add bounds params needed in blex map
+        blex_order_map = add_and_name_isl_dims(
+            blex_order_map, dt.param, blex_order_map_params)
+
+        # Get a set representing blex_order_map space
+        n_blex_dims = n_seq_blex_dims + len(all_par_lex_dim_names)
+        blex_set_template = isl.align_spaces(
+            isl.Map("[ ] -> { [ ] -> [ ] }"), blex_order_map
+            ).move_dims(
+            dt.in_, n_blex_dims, dt.out, 0, n_blex_dims
+            ).domain()
+        blex_set_affs = isl.affs_from_space(blex_set_template.space)
+
+        # {{{ _create_excluded_map_for_iname
+
+        def _create_excluded_map_for_iname(iname, key_lex_tuples):
```

Comment:
Doubly nested? Yikes!

Reply:
I eliminated+inlined `_create_excluded_map_for_iname` since it is only called once. I had encapsulated it to improve code legibility, but as long as the reader is using vim folding, it's still just as legible (perhaps slightly *more* legible, in fact).